app/models/rating.rb

```diff
@@ -3,6 +3,7 @@ class Rating < ActiveRecord::Base
   belongs_to :user
 
   scope :for_event, -> (event) { joins(:proposal).where("proposals.event_id = ?", event.id) }
+  scope :not_withdrawn, -> { joins(:proposal).where("proposals.state != ?", "withdrawn") }
```

Comment:
That looks right. I'd change two things: instead of comparing against the string "withdrawn", reference the constant that defines what that string is. Actually forget about the second thing, it's just the one thing.